### PR TITLE
Add `filter_max_moving_count` operator.

### PR DESCRIPTION
**Pull request by achoum**
Note: The name of the operator still need to be validated.

### CODE REVIEW

File: docs/src/reference/index.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ Check the index on the left for a more detailed description of any symbol.
 | [`EventSet.enumerate()`][temporian.EventSet.enumerate]                                                     | Creates an ordinal feature enumerating the events according to their timestamp.                              |
 | [`EventSet.experimental_fast_fourier_transform()`][temporian.EventSet.experimental_fast_fourier_transform] | Apply a Fast Fourier Transform.                                                                              |
 | [`EventSet.filter()`][temporian.EventSet.filter]                                                           | Filters out events in an [`EventSet`][temporian.EventSet] for which a condition is false.                    |
+| [`EventSet.filter_max_moving_count()`][temporian.EventSet.filter_max_moving_count]                         | Skips events such that no more than `max_count` events are within a time window of `window_length`           |
```
Comment (ianspektor) [inline docs/src/reference/index.md:53]: missing dot at the end

Comment (ianspektor) [reply docs/src/reference/index.md:53]: also update with docstring without `max_count`

File: temporian/core/event_set_ops.py [hunk 2]
```diff
@@ -2420,3 +2420,41 @@ def unique_timestamps(self: EventSetOrNode) -> EventSetOrNode:
...
+    ) -> EventSetOrNode:
+        """Filter out events such that no more than one output event is within a
+        tailing time window of `window_length`.
+
+        Filtering is applied in chronological order: An event received at time t
+        is filtered out if there is a non-filtered out event in
+        (t-window_length, t].
+
+        This operator is different from `(evtset.moving_count(window_length)
+        == 0).filter()`. In `filter_max_moving_count` a filtered event is not
```
Comment (ianspektor) [inline temporian/core/event_set_ops.py:2435]: s/is not/does not

File: temporian/core/event_set_ops.py [hunk 3]
```diff
@@ -2420,3 +2420,41 @@ def unique_timestamps(self: EventSetOrNode) -> EventSetOrNode:
...
+            features: []
+            events:
+                 (2 events):
+                    timestamps: [1. 3.]
+            ...
+
+            ```
+
+        Returns:
+            EventSet without features with unique timestamps in the input.
```
Comment (ianspektor) [inline temporian/core/event_set_ops.py:2454]: s/with unique timestamps in the input/with the filtered events

File: temporian/core/operators/filter_max_moving_count.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+
+@typecheck
+@compile
+def filter_max_moving_count(
+    input: EventSetOrNode, window_length: Duration
+) -> EventSetOrNode:
+    return FilterMaxMovingCount(
+        input=input,
+        window_length=normalize_duration(window_length),
```
Comment (ianspektor) [inline temporian/core/operators/filter_max_moving_count.py:89]: same as until_next, would move normalize_duration to class' constructor

File: temporian/implementation/numpy/operators/test/filter_max_moving_count_test.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+        op = FilterMaxMovingCount(input=evset.node(), window_length=1.5)
+        instance = FilterMaxMovingCountNumpyImplementation(op)
+        testOperatorAndImp(self, op, instance)
+        output = instance.call(input=evset)["output"]
+        assertEqualEventSet(self, output, expected_output)
+
+    def test_like_unique(self):
+        # TODO: Use "math.nextafter" after drop of python 3.8.
+        na = np.nextafter(0, 1)
```
Comment (ianspektor) [inline temporian/implementation/numpy/operators/test/filter_max_moving_count_test.py:69]: should this be np.nextafter(1, 2)? nextafter(0, 1) gives the next fp value after 0

**Review by ianspektor — APPROVED** [review]

File: docs/src/reference/index.md (hunk 1 above)
Comment (achoum) [reply docs/src/reference/index.md:53]: *Done x2

File: temporian/core/operators/filter_max_moving_count.py (hunk 4 above)
Comment (achoum) [reply temporian/core/operators/filter_max_moving_count.py:89]: I'll wait to resolve the discussion on the other PR, and update this PR according :)

File: temporian/implementation/numpy/operators/test/filter_max_moving_count_test.py (hunk 5 above)
Comment (achoum) [reply temporian/implementation/numpy/operators/test/filter_max_moving_count_test.py:69]: >> nextafter(0, 1) gives the next fp value after 0
Yes, and this is the objective => Making sure there is no two events with the same timestamps (like the "unique" operator).

*Added test with p.nextafter(1, 2)

**Pull request merged**